### PR TITLE
Solana Lambda Invoke tutorial

### PR DESCRIPTION
**Pull request by aditya-passbird**

### CODE REVIEW

File: lambda/solana-memo-lambda-invoke/scripts/logMemo.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import { invokeSolana } from "../../lambdaUtils.mjs";
+import {
+  waitUntilTransactionMined,
+  checkAPIKey,
+} from "../../../helpers/utils.mjs";
+
+import logMemoSerializedMessage from "../backend/utils.cjs";
+import getDeveloperWallet from "../../../helpers/utils.mjs";
+
+async function main() {
```
Comment (rohan-passbird) [inline lambda/solana-memo-lambda-invoke/scripts/logMemo.js:11]: doesn't seem to be working
image

File: lambda/solana-memo-lambda-invoke/backend/utils.cjs [hunk 2]
```diff
@@ -0,0 +1,25 @@
+const Web3 = require("@solana/web3.js");
```
Comment (rohan-passbird) [inline lambda/solana-memo-lambda-invoke/backend/utils.cjs:1]: why do we have mix of cjs, mjs and js. Be consistent.

File: lambda/solana-memo-lambda-invoke/README.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
+# MetaKeep Lambda - Invoke Memo Program in Solana
```
Comment (rohan-passbird) [inline lambda/solana-memo-lambda-invoke/README.md:1]: we should instead move all folders inside `solana` to `solana/transaction-signing` and then create a new folder `solana/lambda`

File: lambda/solana-memo-lambda-invoke/scripts/logMemo.js (hunk 1 above)
Comment (rohan-passbird) [reply lambda/solana-memo-lambda-invoke/scripts/logMemo.js:11]: It's the same issue that I fixed earlier this week. `import` or `require` only works in files below the current directory.

File: lambda/solana-memo-lambda-invoke/backend/.env [hunk 4]
```diff
@@ -0,0 +1,6 @@
+// Your MetaKeep API key
+API_KEY =
+
+// Your Lambda contract address.
+// We will be using memo program for solana lambda address
+MEMO_PROGRAM_ADDRESS =
```
[the remark this answers is not in the document]
Comment (rohan-passbird) [reply lambda/solana-memo-lambda-invoke/backend/.env:6]: Looks like this env is not actually being used anywhere

[the remark this answers is not in the document]
Comment (aditya-passbird) [reply lambda/solana-memo-lambda-invoke/backend/.env]: fixed

File: lambda/solana-memo-lambda-invoke/README.md (hunk 3 above)
Comment (aditya-passbird) [reply lambda/solana-memo-lambda-invoke/README.md]: created issue

File: helpers/utils.mjs [hunk 5]
```diff
@@ -29,7 +29,7 @@ export default async function getDeveloperWallet() {
   }
 
   console.log("\n");
-  return resultJson.wallet.ethAddress;
+  return resultJson.wallet.ethAddress || resultJson.wallet.solAddress;
```
Comment (rohan-passbird) [inline helpers/utils.mjs:32]: this seems very confusing. Make 2 functions instead `getDeveloperEvmAddress`, `getDeveloperSolAddress`

Comment (rohan-passbird) [reply helpers/utils.mjs:32]: Same for user functions below `getUserEvmAddress`, `getUserSolAddress`

File: helpers/utils.mjs [hunk 6]
```diff
@@ -178,5 +181,5 @@ export const checkAPIKey = () => {
 };
 
 export const getAPIHost = () => {
-  return process.env.API_HOST || "https://api.metakeep.xyz";
+  return process.env.API_HOST || "https://api.dev.metakeep.xyz";
```
Comment (rohan-passbird) [inline helpers/utils.mjs:184]: undo this

Comment (aditya-passbird) [reply helpers/utils.mjs]: done

File: helpers/utils.mjs (hunk 5 above)
Comment (aditya-passbird) [reply helpers/utils.mjs]: done

Comment (rohan-passbird) [reply helpers/utils.mjs]: We don't need to duplicate the whole function. 99% of the code is the same. Also, see the function names I mentioned above: they end in Address for more clarity.

Comment (aditya-passbird) [reply helpers/utils.mjs]: done

**Review by rohan-passbird — APPROVED** [review]

File: helpers/utils.mjs (hunk 6 above)
Comment (rohan-passbird) [reply helpers/utils.mjs]: Fixed.

**Pull request merged**